src/main/java/mate/jdbc/dao/CarDaoImpl.java

```diff
@@ -0,0 +1,212 @@
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement getAllCarsStatement = connection.prepareStatement(getRequest)) {
+            getAllCarsStatement.setLong(1, driverId);
+            ResultSet resultSet = getAllCarsStatement.executeQuery();
+            List<Car> cars = new ArrayList<>();
+            while (resultSet.next()) {
+                cars.add(getCar(resultSet));
+            }
+            return cars;
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can't find cars in DB by driver driverId "
+                    + driverId, e);
+        }
+    }
+
+    private List<Driver> getDriversForCar(Long carId) {
+        String request = "SELECT id, name, license_number FROM drivers d "
+                + "JOIN cars_drivers cd ON d.id = cd.driver_id WHERE cd.car_id = ?";
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement getAllDriversStatement = connection.prepareStatement(request)) {
+            getAllDriversStatement.setLong(1, carId);
+            ResultSet resultSet = getAllDriversStatement.executeQuery();
+            List<Driver> drivers = new ArrayList<>();
+            while (resultSet.next()) {
+                drivers.add(getDriver(resultSet));
+            }
+            return drivers;
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can't find drivers in DB by car carId " + carId, e);
+        }
+    }
+
+    private Driver getDriver(ResultSet resultSet) {
+        try {
+            Long id = resultSet.getLong("id");
+            String name = resultSet.getString("name");
+            String licenseNumber = resultSet.getString("license_number");
+            return new Driver(id, name, licenseNumber);
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can't get driver from DB", e);
+        }
+    }
+
+    private Car getCar(ResultSet resultSet) throws SQLException {
+        Long manufacturerId = resultSet.getLong("manufacturer_id");
+        String manufacturerName = resultSet.getString("name");
+        String manufacturerCountry = resultSet.getString("country");
+        Manufacturer manufacturer =
+                new Manufacturer(manufacturerId, manufacturerName, manufacturerCountry);
+        Long id = resultSet.getLong("id");
+        String model = resultSet.getString("model");
+        return new Car(id, model, manufacturer);
+    }
+
+    private void insertDriverToDriverDB(List<Driver> drivers) {
+        String query = "INSERT INTO drivers (id, name, license_number) VALUES(?, ?, ?)";
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement statement = connection.prepareStatement(query,
+                        Statement.RETURN_GENERATED_KEYS)) {
+            for (Driver driver : drivers) {
+                statement.setLong(1, driver.getId());
```

Comment:
The ID should be generated automatically
```suggestion
```

Reply:
not fixed